### PR TITLE
b UI 0001 validate version of file

### PR DESCRIPTION
**Pull request by rui-abrantes**
Creating endpoint for uploading the .xlsx and .csv extensions + Acceptance Criterias
        - File Acceptance
	- Verification Validation
	- Confirmation Message
	- Logging (both success and failure)

### CODE REVIEW

**Review by luismonteiro98 — APPROVED** [review]

File: BoardUserInterface.API/UploadFiles/ExcelMetadataService.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+namespace BoardUserInterface.API.UploadFiles;
+
+public interface IExcelMetadataService
+{
+    string GetVersionNumberFromExcel(Stream excelFileStream);
+}
+
+public class ExcelMetadataService : IExcelMetadataService
+{
+    /*
```
Comment (hataide) [inline BoardUserInterface.API/UploadFiles/ExcelMetadataService.cs:15]: You should remove all the comments

**Review by hataide — CHANGES_REQUESTED** [review]

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        _logger = logger;
+        _excelMetadataService = excelMetadataService;
+        _fileTemplateInformations = new List<FileTemplateInformation>(); // Replace with your actual storage mechanism
+    }
+
+    public async Task<string> UploadFileAsync(IFormFile file)
+    {
+        if (file == null || file.Length == 0)
+        {
+            //_logger.LogError("Error during file upload. No file was uploaded or the file is empty.");
```
Comment (hataide) [inline BoardUserInterface.API/UploadFiles/FileUploadService.cs:29]: More comments to remove

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            throw new ArgumentException("File is empty or null.");
+        }
+
+        var fileName = ContentDispositionHeaderValue.Parse(file.ContentDisposition).FileName.Trim('"');
+        var fileExtension = Path.GetExtension(fileName).ToLowerInvariant();
+
+        if (fileExtension != ".xlsx" && fileExtension != ".csv")
+        {
+            //_logger.LogError($"Error during file upload. Invalid file extension: {fileExtension}. Only .xlsx and .csv files are allowed.");
```
Comment (hataide) [inline BoardUserInterface.API/UploadFiles/FileUploadService.cs:38]: More comments to remove

File: BoardUserInterface.API/UploadFiles/ExcelMetadataService.cs (hunk 1 above)
Comment (rui-abrantes) [reply BoardUserInterface.API/UploadFiles/ExcelMetadataService.cs]: I removed all the comments!

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+public interface IFileUploadService
+{
+    Task<string> UploadFileAsync(IFormFile file);
+}
+
+public class FileUploadService : IFileUploadService
+{
+    private readonly ILogger<FileUploadService> _logger;
+    private readonly IExcelMetadataService _excelMetadataService;
+    private readonly List<FileTemplateInformation> _fileTemplateInformations; // This should be replaced with your actual storage mechanism
```
Comment (hataide) [inline BoardUserInterface.API/UploadFiles/FileUploadService.cs:16]: There is no need for this kind of comments

**Review by hataide — CHANGES_REQUESTED** [review]

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs (hunk 4 above)
Comment (rui-abrantes) [reply BoardUserInterface.API/UploadFiles/FileUploadService.cs]: Removed it

**Review by hataide — CHANGES_REQUESTED** [review]

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    public FileUploadService(ILogger<FileUploadService> logger, IExcelMetadataService excelMetadataService)
+    {
+        _logger = logger;
+        _excelMetadataService = excelMetadataService;
+        _fileTemplateInformations = new List<FileTemplateInformation>(); // Replace with your actual storage mechanism
```
[the remark this answers is not in the document]
Comment (rui-abrantes) [reply BoardUserInterface.API/UploadFiles/FileUploadService.cs]: Did it

File: BoardUserInterface.API/UploadFiles/FileUploadService.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        {
+            using (var memoryStream = new MemoryStream())
+            {
+                await file.CopyToAsync(memoryStream);
+                memoryStream.Position = 0; // Reset the memory stream position after copying
+                versionNumber = _excelMetadataService.GetVersionNumberFromExcel(memoryStream);
+            }
+        }
+
+        // Create an instance of FileTemplateInformation and add it to the list
```
[the remark this answers is not in the document]
Comment (rui-abrantes) [reply BoardUserInterface.API/UploadFiles/FileUploadService.cs]: Removed it

**Review by hataide — APPROVED** [review]

**Pull request merged**